### PR TITLE
Few Shot Chat Prompt

### PR DESCRIPTION
**Pull request by hinthornw**
Proposal for a few shot chat message example selector

### CODE REVIEW

**Comment by eyurtsev** [comment]
@willsmithorg let me know the change is okay -- expanded the doc-string in attempt to explain the high level structure and the purpose and added a simple example that uses static examples. I can update the documentation for chat messages in the notebooks.

**Comment by hinthornw** [comment]
@eyurtsev Think you tagged the wrong @hinthornw

File: libs/langchain/langchain/prompts/few_shot.py [hunk 1]
```diff
@@ -167,64 +167,118 @@ def dict(self, **kwargs: Any) -> Dict:
...
-            llm=ChatOpenAI(),
-            prompt=few_shot_prompt,
-        )
-        chain({"input": "What's 3+3?"})
+    This prompt template can be used to generate a fixed list of examples or else
+    to dynamically select examples based on the input.
+
+    Examples:
+
+        Prompt template with fixed a fixed list of examples (matching the sample
```
Comment (hinthornw) [inline libs/langchain/langchain/prompts/few_shot.py:189]:
```suggestion
        Prompt template with a fixed list of examples (matching the sample
```

File: libs/langchain/langchain/prompts/few_shot.py [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 )
+from langchain.prompts.chat import BaseChatPromptTemplate, BaseMessagePromptTemplate
 from langchain.prompts.example_selector.base import BaseExampleSelector
 from langchain.prompts.prompt import PromptTemplate
+from langchain.schema.messages import BaseMessage, get_buffer_string
+from langchain.schema.prompt_template import BasePromptTemplate
 
 
-class FewShotPromptTemplate(StringPromptTemplate):
+class _FewShotPromptTemplateMixin(BaseModel):
```
Comment (baskaryan) [inline libs/langchain/langchain/prompts/few_shot.py:20]: does mixin need to be BaseModel (seems like everything that uses it already is)

Comment (hinthornw) [reply libs/langchain/langchain/prompts/few_shot.py:20]: It was giving some weird metaclass resolution things without but I'll double check that still is the case

File: libs/langchain/langchain/prompts/few_shot.py [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
     """Prompt template that contains few shot examples."""
 
-    @property
-    def lc_serializable(self) -> bool:
-        return False
-
     examples: Optional[List[dict]] = None
```
Comment (hinthornw) [inline libs/langchain/langchain/prompts/few_shot.py:23]: I'm not super convinced this abstraction is useful for static examples fwiw

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**